### PR TITLE
feat:lesson06

### PR DESCRIPTION
**Pull request by nakagawamai**
# [Issue6](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#6)
## Implementation questions
- I would like to know if there is an easier way to write.

### CODE REVIEW

File: lesson06/script.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const lists = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+];
+
+const promise = new Promise(resolve => {
+    setTimeout(() => resolve(lists),3000);
+});
```
Comment (nakagawamai) [inline lesson06/script.js]: I implemented using setTimeout.
I learned that the function specified by setTimeout is called only once.

Comment (seeeen044) [inline lesson06/script.js]:
```suggestion
const promise = new Promise(resolve => {
    const lists = [
    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
];

    setTimeout(() => resolve(lists),3000);
});
```
Nits
I can summarize it.

**Review by seeeen044 — APPROVED** [review]

File: lesson06/script.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-];
-
 const promise = new Promise(resolve => {
+    const lists = [
+        {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+        {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+    ];
+    
     setTimeout(() => resolve(lists),3000);
 });
```
Comment (nakagawamai) [inline lesson06/script.js]: I put a variable declaration in a Promise.
This is to make the code easier to read.

File: lesson06/script.js (hunk 1 above)
Comment (nakagawamai) [reply lesson06/script.js]: Thank you for your comment!
Put the required code in Promise and
Very easy to read!
I will try to summarize.

File: lesson06/script.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const promise = new Promise(resolve => {
```
Comment (taka0209) [inline lesson06/script.js]: nits
I felt it would be better to define a meaningful name.

File: lesson06/script.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
-promise.then((value) => {
+createList.then((value) => {
     const fragment = document.createDocumentFragment();
     for(const list of value){
-        const li = document.createElement('li');
+     const li = document.createElement('li');
         const a = document.createElement('a');
         const img = document.createElement('img');
         a.textContent = list.text;
```
Comment (nakagawamai) [inline lesson06/script.js:16]: I change variable name.
Because meaningful names are easy to understand.

File: lesson06/script.js [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
         {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
         {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
     ];
-    
+
     setTimeout(() => resolve(lists),3000);
 });
 
-promise.then((value) => {
+createList.then((value) => {
```
Comment (nakagawamai) [inline lesson06/script.js:10]: I change variable name.
Because meaningful names are easy to understand.

File: lesson06/script.js [hunk 6]
```diff
@@ -1,16 +1,16 @@
-const promise = new Promise(resolve => {
+const createList = new Promise(resolve => {
```
Comment (nakagawamai) [inline lesson06/script.js:1]: I change variable name.
Because meaningful names are easy to understand.

File: lesson06/script.js (hunk 4 above)
Comment (nakagawamai) [inline lesson06/script.js:16]: I change variable name.
Because meaningful names are easy to understand.

Comment (taka0209) [reply lesson06/script.js:16]: I'll post a link that might be helpful.
https://qiita.com/KeithYokoma/items/2193cf79ba76563e3db6

**Review by taka0209 — APPROVED** [review]

File: lesson06/script.js (hunk 4 above)
Comment (nakagawamai) [reply lesson06/script.js:16]: Thank you for teaching me.
I'll try reading it!

**Pull request merged**